libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -63,39 +63,42 @@ func newHostMetadataProcessor(cfg *common.Config) (processors.Processor, error)
 	p := &addHostMetadata{
 		info:   h.Info(),
 		config: config,
+		data:   common.NewMapStrPointer(nil),
 	}
 	return p, nil
 }
 
 // Run enriches the given event with the host meta data
 func (p *addHostMetadata) Run(event *beat.Event) (*beat.Event, error) {
 	p.loadData()
-	event.Fields.DeepUpdate(p.data.Clone())
+	event.Fields.DeepUpdate(p.data.Get().Clone())
 	return event, nil
 }
 
 func (p *addHostMetadata) loadData() {
-
 	// Check if cache is expired
-	if p.lastUpdate.Add(cacheExpiration).Before(time.Now()) {
-		p.data = host.MapHostInfo(p.info)
-
-		if p.config.NetInfoEnabled {
-			// IP-address and MAC-address
-			var ipList, hwList, err = p.getNetInfo()
-			if err != nil {
-				logp.Info("Error when getting network information %v", err)
-			}
+	if p.lastUpdate.Add(cacheExpiration).After(time.Now()) {
```

Comment:
Use an atomic for p.lastUpdate and:
```suggestion
	lastUpdate := p.lastUpdate.Load()
	if lastUpdate.Add(cacheExpiration).After(time.Now())) || !p.lastUpdate.CAS(lastUpdate, time.Now()) {
```

This way it ensures that only one thread will generate the map once it's expired.
(Yes I have a fixation with atomic operations today)

Reply:
Sounds good to me

Reply:
Well, thinking twice, compare and swap on a `time.Time` can be a bit uglier, would it be much worse to do it with a mutex instead of using atomic? (See 4e99e5864)